### DISCUSSION TITLE
Text after certain character disappears

### DISCUSSION THREAD

**Discussion by aoyama-val** [noborus/ov#384] [Q&A]
Text after " ◯" disappears.

[sample2.txt](https://github.com/noborus/ov/files/11643502/sample2.txt)

```
$ cat sample2.txt
 column1 |  column2
---------+------------
 ◯ほげ   | 2023-06-04
 b       | 2023-06-05

$ ov --config /dev/null sample2.txt
```

![スクリーンショット 2023-06-04 3 55 11](https://github.com/noborus/ov/assets/13144822/d5fb0f43-c175-4b99-8a46-d4880f303a47)

Deleted "◯" from the file, and "ほげ" was displayed correctly, so I guess "◯" was the culprit.
The terminal emulator is iTerm2 and this problem occurs only when `Ambiguous characters are double-width` is checked.
![スクリーンショット 2023-06-04 4 04 49](https://github.com/noborus/ov/assets/13144822/117fb180-da8d-4238-9476-8cda64113449)
When it is not checked, it is displayed as follows:
![スクリーンショット 2023-06-04 4 04 33](https://github.com/noborus/ov/assets/13144822/de45f5c0-650d-455c-b278-41125d3415f0)

**Comment by noborus** [comment]
If you check Ambiguous characters are double-width, also set RUNEWIDTH_EASTASIAN=1.

```
export RUNEWIDTH_EASTASIAN=1
ov sample2.txt
```

I'd like to do this automatically, but I don't know how.
However, if you match this environment variable with the terminal, it will work correctly with many other TUIs.

**Comment by aoyama-val** [comment]
Thanks for the answer. It's working!
But I wonder why go-runewidth couldn't detect from environment variables that my terminal is CJK.
https://github.com/mattn/go-runewidth/blob/2c6a438f68cfe01255a90824599da41fdf76d1e2/runewidth_posix.go#L63
```console
$ locale
LANG="ja_JP.UTF-8"
LC_COLLATE="C"
LC_CTYPE="ja_JP.UTF-8"
LC_MESSAGES="ja_JP.UTF-8"
LC_MONETARY="ja_JP.UTF-8"
LC_NUMERIC="ja_JP.UTF-8"
LC_TIME="ja_JP.UTF-8"
LC_ALL=
```
Anyway please close the issue.

**Comment by noborus** [comment]
This seems to be an issue for others as well so moving to discussion.
